components/post_view/post_list/index.jsx

```diff
@@ -0,0 +1,448 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+
+import LoadingScreen from 'components/loading_screen.jsx';
+import CreateChannelIntroMessage from 'components/post_view/channel_intro_message';
+import Constants from 'utils/constants.jsx';
+import {getNewestPostIdFromPosts, getOldestPostIdFromPosts} from 'utils/post_utils';
+
+import PostList from './post_list';
+
+const POSTS_PER_PAGE = Constants.POST_CHUNK_SIZE / 2;
+
+export default class PostListWrapper extends React.PureComponent {
+    static propTypes = {
+
+        /**
+         * Array of posts in the channel, ordered from newest to oldest
+         */
+        posts: PropTypes.array,
+
+        /**
+         * The channel the posts are in
+         */
+        channel: PropTypes.object.isRequired,
+
+        /**
+         * Whether to display the channel intro at full width
+         */
+        fullWidth: PropTypes.bool,
+
+        /**
+         * Used for disabling loading more messages
+         */
+        postVisibility: PropTypes.number,
+
+        /**
+         * Used for showing more message indicators and channel intro message
+         */
+        channelPostsStatus: PropTypes.object,
+
+        /**
+         * Used for unread count
+         */
+        currentUserId: PropTypes.string,
+
+        /**
+         * Used for unread count
+         */
+        lastViewedAt: PropTypes.number,
+
+        /**
+         * Used for getting lastViewed timestamp from api.
+         */
+        member: PropTypes.object,
```

Comment:
Might not need the full `member` object.  Why not use the existing `lastViewedAt`?  Or maybe process the logic in the container?

Reply:
This logic will change. For now I will remove the dependency of this and figure out a way in another PR for unread.